pg_search/src/postgres/customscan/dsm.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2023-2024 Retake, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use crate::postgres::customscan::{CustomScan, ParallelQueryCapable};
+use pgrx::{pg_guard, pg_sys};
+
+/// Estimate the amount of dynamic shared memory that will be required for parallel operation. This
+/// may be higher than the amount that will actually be used, but it must not be lower. The return
+/// value is in bytes. This callback is optional, and need only be supplied if this custom scan
+/// provider supports parallel execution.
+#[pg_guard]
+pub extern "C" fn estimate_dsm_custom_scan<CS: CustomScan + ParallelQueryCapable>(
+    node: *mut pg_sys::CustomScanState,
+    pcxt: *mut pg_sys::ParallelContext,
+) -> pg_sys::Size {
+    todo!("estimate_dsm_custom_scan")
```

Comment:
Are all of these TODOs part of V2? I wonder if it's approachable enough to create GitHub Issues for others to pick from in parallel, if so.

Reply:
I'm changing these to `unimplemented!`.

This is not something I'd expect the community to do.  This "DSM" stuff is about parallel custom scans.  If we support that (and we probably will at some point), we'll do it ourselves.

Reply:
Sounds good